griptape/tools/google_sheets/tool.py

```diff
@@ -0,0 +1,436 @@
+        from google.auth.exceptions import MalformedError
+        from googleapiclient.errors import HttpError
+
+        values = params["values"]
+        file_type = values.get("file_type")
+        file_path = values.get("file_path")
+        file_name = values.get("file_name")
+
+        mime_mapping = {
+            "csv": "text/csv",
+            "excel": "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+        }
+
+        if file_type not in mime_mapping:
+            logging.error(f"Unsupported file type '{file_type}' provided.")
+            return ErrorArtifact(
+                f"Unsupported file type '{file_type}'. Please provide either 'csv' or 'excel'."
+            )
+
+        try:
+            service = self._build_client(
+                scopes=self.DRIVE_UPLOAD_SCOPES,
+                service_name="drive",
+                version="v3",
+                owner_email=self.owner_email,
+            )
+
+            file_metadata = {
+                "name": file_name,
+                "mimeType": "application/vnd.google-apps.spreadsheet",
+            }
+            media = MediaFileUpload(
+                file_path, mimetype=mime_mapping[file_type]
+            )
+            file = (
+                service.files()
+                .create(body=file_metadata, media_body=media, fields="id")
+                .execute()
+            )
+
+            return InfoArtifact(
+                f'file was successfully uploaded and converted to a Google Sheet with ID {file.get("id")}.'
+            )
+        except HttpError as e:
+            return ErrorArtifact(
+                f"error uploading spreadsheet due to http error: {e}"
+            )
+        except MalformedError as e:
+            return ErrorArtifact(
+                f"error uploading spreadsheet due to malformed credentials: {e}"
+            )
+        except Exception as e:
+            return ErrorArtifact(
+                f"error uploading and converting file to Google Sheet: {e}"
+            )
+
+    @activity(
+        config={
+            "description": "Can be used to share a spreadsheet with a specified user.",
+            "schema": Schema(
+                {
+                    Literal(
+                        "file_path",
+                        description="The path of the spreadsheet to share",
+                    ): str,
+                    Literal(
+                        "email_address",
+                        description="The email address of the user to share with",
+                    ): str,
+                    Optional(
+                        "role",
+                        default="reader",
+                        description="The role to give to the user, e.g., 'reader', 'writer', or 'commenter'",
+                    ): Or("reader", "writer", "commenter"),
+                }
+            ),
+        }
+    )
+    def share_spreadsheet(self, params: dict) -> InfoArtifact | ErrorArtifact:
```

Comment:
Should this live in the Drive tool, so that any file can be shared? Or can this only be applied to Docs and Sheets? If so, we should move this activity into a mixin and make it work for the Docs and Sheets tools.

Reply:
Yup! Moved it to drive!